ClothesApp/WebAPI/Controllers/BrandsController.cs

```diff
@@ -1,57 +1,101 @@
-using Application.Dtos;
-using Application.Services;
+using Application.Dtos.Brands;
+using Application.Dtos.Products;
+using Application.Interfaces.Services;
 using Microsoft.AspNetCore.Mvc;
 
 namespace WebAPI.Controllers;
 
 [ApiController]
-[Route("api/Brands")] 
+[Route("api/brands")] 
 public class BrandsController : ControllerBase
 {
-    private readonly IBrandService _brandService;   
+    private readonly IBrandService _brandService;  
     
-    public BrandsController(IBrandService brandService)
+    private readonly IProductService _productService;  
+    
+    public BrandsController(IBrandService brandService, IProductService productService)
     {
-       _brandService = brandService;
+        _brandService = brandService;
+        _productService = productService;
     }
 
     [HttpGet]
-    public async Task<ActionResult<IReadOnlyCollection<BrandDto>>> GetAllBrands()
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<BrandDto>))]
+    public async Task<ActionResult<IList<BrandDto>>> GetAllBrands()
     {
-        var brandsDtos = await _brandService.GetAllBrands();
+        var brandDtos = await _brandService.GetAll();
 
-        return Ok(brandsDtos);
+        return Ok(brandDtos);
+    }
+    
+    [HttpGet("{id}/products")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<ProductDto>))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult<IList<ProductDto>>> GetBrandProducts([FromRoute] long id)
+    {
+        var productDtos = await _productService.GetProductsByBrandId(id);
+
+        return Ok(productDtos);
     }
     
     [HttpGet("{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BrandDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
     public async Task<ActionResult<BrandDto>> GetBrandById([FromRoute] long id)
```

Comment:
What will be the case for getting single brand by id?
I want you to think about all use cases for your endpoints if it's the real eshop, some endpoints might be not needed

Reply:
It can be the brand page. It's redundant